src/main/java/eu/miaplatform/customplugin/ServiceClientFactory.java

```diff
@@ -17,4 +21,8 @@ public static CRUDServiceClient getCRUDServiceClient(String apiPath, int version
     public static CRUDServiceClient getCRUDServiceClient(String apiPath, String apiSecret, int version, CustomPluginHeadersPropagator headersPropagator) {
         return new CRUDServiceClientImpl(apiPath, apiSecret, version, headersPropagator);
     }
+
+    public static Service getServiceProxy(String serviceName, JsonObject requestMiaHeaders, InitServiceOptions options) {
```

Comment:
I'd create two methods with different signatures: one with `requestMiaHeaders` and one without. 

Also I would not use `JsonObject` to avoid coupling with gson library; I'd define a custom class built on top of it (or built on top of okhttp3 headers).

I've seen okhttp3 and they use a simple `Iterable<Pair<String,String>>` we could use the same approach (so a custom class that extends  `Iterable<Pair<String,String>>` (or okhttp3 headers directly).

Reply:
For what concerns `JsonObject` coupling, I suppose it's the same for `Service` class methods where a request body in JSON format is needed as parameter. What shall we use? Simply using `String` could be ok?